utils: prepare for enabling the dynamic experimental SDK

compnerd opened the pull request:
This series of changes prepare us to build the dynamic experimental SDK. We currently are missing:
1. Differentiation
2. SwiftRemoteMirror
3. C++ Interop Overlays

Beyond this, we have a complete SDK built statically. We want to start preparing the dynamic library variant so that we can package the experimental SDK and begin the transition.

compnerd committed:
utils: simplify signature for `Build-ExperimentalRuntime`

compnerd committed:
utils: rename `StaticFoundation` to `ExperimentalFoundation`

compnerd committed:
utils: specify linkage types for experimental SDK

compnerd committed:
utils: precompute binary cache locations

compnerd commented:
@swift-ci please smoke test

compnerd commented:
@swift-ci please smoke test

compnerd commented:
CC: @Steelskin

compnerd commented:
@swift-ci please smoke test

Steelskin commented on utils/build.ps1:
```diff
@@ -2555,53 +2583,53 @@ function Build-ExperimentalRuntime {
     Build-CMakeProject `
       -Src $SourceCache\swift\Runtimes\Supplemental\Differentiation `
-      -Bin (Get-ProjectBinaryCache $Platform ExperimentalDifferentiation) `
+      -Bin (Get-ProjectBinaryCache $Platform ExperimentalStaticDifferentiation) `
```
Did you miss that one or is it intended to always be built statically? If the latter, please add a comment and remove the `BUILD_SHARED_LIBS` configuration.

compnerd committed:
build.ps1: compute SDKROOT once in the experimental runtimes build

This value is re-used. Compute it once and re-use the value to ease the
future transition.
